src/pagination/Pagination.js

```diff
@@ -3,25 +3,36 @@ import { PropTypes } from '@dhis2/prop-types'
 import i18n from '@dhis2/d2-i18n'
 import { useHistory } from 'react-router-dom'
 import { Button, SingleSelect, SingleSelectOption } from '@dhis2/ui'
-import { useQueryParams } from './useQueryParams'
-import { createSearchString } from './createSearchString'
+import { useQueryParams } from '../hooks'
+import { createSearchString } from '../utils'
 import styles from './Pagination.module.css'
 
 const PAGE_LENGTHS = ['1', '2', '10', '20', '30', '40', '50', '100']
 
-const Pagination = ({ page, pageCount, pageSize, total }) => {
+const take = (object, keys) => {
+    const o = {}
+    for (const key of Object.keys(object)) {
+        if (keys.includes(key)) {
+            o[key] = object[key]
+        }
+    }
+    return o
+}
+
+const Pagination = ({ pager, extraSearchParams = [] }) => {
+    const { page, pageCount, pageSize, total } = pager
     const firstItem = Math.min((page - 1) * pageSize + 1, total)
     const lastItem = Math.min(firstItem + pageSize - 1, total)
     const availablePages = Array.from({ length: pageCount }, (_x, i) =>
         (i + 1).toString()
     )
-    const { status, phoneNumber } = useQueryParams()
+    const queryParams = useQueryParams()
     const history = useHistory()
     const navigateToPage = newPage => {
         history.push({
             search: createSearchString({
-                status,
-                phoneNumber,
+                status: queryParams.status,
+                ...take(queryParams, extraSearchParams),
```

Comment:
If you spread the queryParams object here you should be able to just overwrite the params that need to be changed afterwards right? Or am I overlooking something?

Reply:
You're right that this isn't needed - I thought I'd need this because `useQueryParams` returns `phoneNumber` regardless of whether it's actually used and I didn't want a `phoneNumber` query param to appear on the sent SMS list but it turns out that `createSearchString` filters out empty strings.

Reply:
I see. Assuming that we'll also have a phone number filter on the sent route we should then be able to simplify this area a bit.